@typescript-eslint rules are applied to javscript files

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
<!-- What is the behavior that currently you experience? -->

This is a contrived example to demonstrate lint problems that emerge when installing a storybook configuration on a react project.

When I create a javascript file in a typescript (react) project, for example:
```
const foo = require('react');
```

Then `eslint` fails with the following rule: 
```
Require statement not part of import statement.(@typescript-eslint/no-var-requires)
```

Since the `@typescript-eslint` plugin is only used in the `@nrwl/nx/typescript` configuration, and in the base `.eslintrc.json` is only applied to files of type `*.ts` and `*.tsx`, I believe this is a bug. More, if typescript rules are needlessly being run on javascript files, then this is an overhead.

## Expected Behavior
<!-- What is the behavior that you expect to happen? -->
<!-- Is this a regression? .i.e Did this used to be the behavior at one point?  -->

`@typescript-eslint` rules to be ignored for `*.js` and `*.jsx` files.

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->

<!-- Can you reproduce this on https://github.com/nrwl/nx-examples? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->

`yarn create nx-workspace --preset=react`

Then create a file `blah.js` at `./apps/app/src/app` and add `const foo = require('react')` to it.

Run `yarn lint`. You should see:
```
  1:7   warning  'foo' is assigned a value but never used        @typescript-eslint/no-unused-vars
  1:13  error    Require statement not part of import statement  @typescript-eslint/no-var-requires
```

### Failure Logs
<!-- Please include any relevant log snippets or files here. -->

### Environment
<!-- It's important for us to know the context in which you experience this behavior! -->
<!-- Please paste the result of `nx report` below! -->

```
  Node : 14.14.0
  OS   : darwin x64
  npm  : 6.14.8
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 11.1.5
  @nrwl/cypress : 11.1.5
  @nrwl/devkit : 11.1.5
  @nrwl/eslint-plugin-nx : 11.1.5
  @nrwl/express : Not Found
  @nrwl/jest : 11.1.5
  @nrwl/linter : 11.1.5
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : 11.1.5
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.1.5
  @nrwl/web : 11.1.5
  @nrwl/workspace : 11.1.5
  typescript : 4.0.5
```

--- a/packages/eslint-plugin-nx/src/configs/javascript.ts
+++ b/packages/eslint-plugin-nx/src/configs/javascript.ts
@@ -33,5 +33,11 @@ export default {
     '@typescript-eslint/explicit-module-boundary-types': 'off',
     '@typescript-eslint/explicit-function-return-type': 'off',
     '@typescript-eslint/no-parameter-properties': 'off',
+    /**
+     * Until ESM usage in Node matures, using require in e.g. JS config files
+     * is by far the more common thing to do, so disabling this to avoid users
+     * having to frequently use "eslint-disable-next-line" in their configs.
+     */
+    '@typescript-eslint/no-var-requires': 'off',
   },
 };

